json_serializable/test/config_test.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:build/build.dart';
+import 'package:json_serializable/builder.dart';
+import 'package:logging/logging.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:test/test.dart';
+
+final _throwsCastError = throwsA(const isInstanceOf<CastError>());
+
+void main() {
+  StreamSubscription sub;
+  final records = <LogRecord>[];
+
+  setUpAll(() {
+    sub = log.onRecord.listen(records.add);
+  });
+
+  tearDownAll(() async {
+    await sub.cancel();
+  });
+
+  setUp(records.clear);
+
+  test('empty', () async {
+    var builder = jsonSerializable(BuilderOptions.empty) as PartBuilder;
+    expect(builder, isNotNull);
+    expect(records, isEmpty);
+  });
+
+  test('valid config', () async {
+    var builder =
+        jsonSerializable(const BuilderOptions(_validConfig)) as PartBuilder;
+    expect(builder, isNotNull);
+
+    expect(records, isEmpty);
+  });
+
+  test('unsupported configuration', () async {
+    var builder =
+        jsonSerializable(const BuilderOptions(const {'unsupported': 'config'}))
+            as PartBuilder;
+    expect(builder, isNotNull);
+
+    expect(records.single.message,
```

Comment:
I don't get how this is passing given my comment above...

Reply:
as discussed...